Obsidian/WorldData/World.cs

```diff
@@ -279,6 +281,60 @@ public IEnumerable<Entity> GetEntitiesNear(VectorF location, float distance = 10
 
     public bool RemovePlayer(Player player) => this.Players.TryRemove(player.Uuid, out _);
 
+    /// <summary>
+    /// Method that handles world-specific tick behavior.
+    /// </summary>
+    /// <returns></returns>
+    public async Task DoWorldTickAsync()
+    {
+        this.Data.Time += this.Server.Config.TimeTickSpeedMultiplier;
+        this.Data.RainTime -= this.Server.Config.TimeTickSpeedMultiplier;
+
+        if (Data.RainTime < 1)
+        {
+            // Raintime passed, toggle weather
+            Data.Raining = !Data.Raining;
+
+            int rainTime;
+            // amount of ticks in a day is 24000
+            if (Data.Raining)
+            {
+                rainTime = Globals.Random.Next(12000, 24000); // rain lasts 0.5 - 1 day
+            }
+            else
+            {
+                rainTime = Globals.Random.Next(12000, 180000); // clear lasts 0.5 - 7.5 day
+            }
+            Data.RainTime = rainTime;
```

Comment:
Could we do something like:
```csharp
// Rain lasts 0.5-1 day. Clear lasts 0.5 - 7.5 days.
Data.RainTime = Globals.Random.Next(TICKS_DAY/2, Data.Raining ? TICKS_DAY : (int)(TICKS_DAY * 7.5));
```
or is that too convoluted?

Reply:
I had the same idea, but it indeed is convoluted.

Reply:
Right. Readability > Concise.

Reply:
I'll think about it. I do like the idea of making it a constant.

Reply:
I'll leave it as-is considering the conversation is resolved